7_react/react2/week1/vite-project/src/App.jsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState, useContext } from "react";
+import "./App.css";
+
+// Create a new context
+const GithubContext = React.createContext();
+
+// Create a custom hook to use the GithubContext
+export const useGithubContext = () => {
+  return useContext(GithubContext);
+};
+
+// Create a GithubProvider component to manage the state
+export const GithubProvider = ({ children }) => {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(null);
+  const [users, setUsers] = useState([]);
+
+  const searchUsers = async (query) => {
+    setLoading(true);
+    setError(null);
+
+    try {
+      const response = await fetch(
+        `https://api.github.com/search/users?q=${query}`
+      );
+      const data = await response.json();
+
+      if (response.ok) {
+        setUsers(data.items);
+      } else {
+        setError(data.message);
+      }
+    } catch (error) {
+      setError(error.message);
+    }
+
+    setLoading(false);
+  };
+
+  return (
+    <GithubContext.Provider value={{ loading, error, users, searchUsers }}>
+      {children}
+    </GithubContext.Provider>
+  );
+};
+
+// Create a Search component
+const Search = () => {
```

Comment:
Please move the Search component to a separate file

Reply:
done